### PR TITLE
Feature/ele 502 create new assignment in plan

### PR DESCRIPTION
**Pull request by dannelundqvist**
Main part of creating new assignments as well as editing existing assignments. Fixed issues in using Textbit as textbox.

### CODE REVIEW

File: src/views/Planning/components/AssignmentRow.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+          setSelectedAssignment={setSelectedAssignment}
+          setShowVerifyDialog={setShowVerifyDialog}
+        />
+      }
+    </div>
+  )
+}
+
+
+function VerifyDeleteAssignmentDialog({ index, title, setSelectedAssignment, setShowVerifyDialog }: {
```
Comment (gularsson) [inline src/views/Planning/components/AssignmentRow.tsx:90]: Picky, but shouldn't it be `const VerifyDeleteAssignmentDialog () =>` for style reasons

**Review by gularsson — CHANGES_REQUESTED** [review]
Amazing work. This is gonna be nice to have up and running!

File: src/lib/planning/createPlanningAssignment.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  assignments.push([assignment])
+}
+
+/**
+ * Create a template structure for an assigment
+ *
+ * TODO: Should be refactored into a more coherent group of functions with createPlanningDocument etc
+ * TODO: Should return combination of Block and Record<string, Block[]>
+ */
+function getAssignmentTemplate(id: string, assignmentType: string): Y.Map<unknown> {
```
[the remark this answers is not in the document]
Comment (dannelundqvist) [reply src/lib/planning/createPlanningAssignment.ts:36]: Forgot to circle back to that one. It's definitely wrong.

File: src/lib/planning/createPlanningDocument.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+/**
+ * Create a template structure for a aplanning
+ *
+ * TODO: Should be refactored into a more coherent group of functions with createPlanningAssignment etc
+ */
+function getPlanningTemplate(id: string): Document {
+  return {
+    // @ts-expect-error TODO: Remove this line when merged with snapshot-functionality branch
```
[the remark this answers is not in the document]
Comment (dannelundqvist) [reply src/lib/planning/createPlanningDocument.ts:28]: Haven't merged it yet.

**Review by gularsson — APPROVED** [review]

**Pull request merged**